presto-main/src/main/java/com/facebook/presto/type/VarcharOperators.java

```diff
@@ -150,6 +150,19 @@ public static double castToDouble(@SqlType("varchar(x)") Slice slice)
 
     @LiteralParameters("x")
     @ScalarOperator(CAST)
+    @SqlType(StandardTypes.FLOAT)
+    public static long castToFloat(@SqlType("varchar(x)") Slice slice)
```

Comment:
Should we introduce `FloatLiteral` class as well and than directly create it in `AstBuilder#visitTypeConstructor`?